lib/spack/spack/spec.py

```diff
@@ -4312,19 +4542,22 @@ def splice(self, other, transitive):
         nodes.update(self_nodes)
 
         for name in nodes:
+            # TODO: check if splice semantics is respected
```

Comment:
I'm guessing the intent would be to resolve this TODO before merging this.

What is the effect of splicing in a dependency when there are multiple instances of the package in the DAG? I would assume that this function might need to specify `deptypes` in that case.

Reply:
> I'm guessing the intent would be to resolve this TODO before merging this.

To my understanding splicing is not yet used in Spack and there's just been some commit to prepare for this functionality. We can either leave a TODO and check it in a later PR (since there are no test failures) or see if @becker33 or maybe @nhanford can help with that here.